cmd/common.go

```diff
@@ -137,11 +139,32 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 			return fmt.Errorf("reading Kong version: %w", err)
 		}
 	}
-	parsedKongVersion, err = parseKongVersion(kongVersion)
+	parsedKongVersion, err = utils.ParseKongVersion(kongVersion)
 	if err != nil {
 		return fmt.Errorf("parsing Kong version: %w", err)
 	}
 
+	if parsedKongVersion.GTE(utils.Kong300Version) &&
+		targetContent.FormatVersion != formatVersion30 {
+		formatVersion := targetContent.FormatVersion
+		if formatVersion == "" {
+			formatVersion = defaultFormatVersion
+		}
+		return fmt.Errorf("cannot apply '%s' format config to Kong instance running\n"+
+			"a release higher or equal to 3.0.\n"+
+			"Please upgrade your configuration format before applying any change.\n"+
+			"This can be achieved via the `deck convert` sub-command:\n\n"+
+			"deck convert --from kong-gateway-2.x --to kong-gateway-3.x <opts>\n\n"+
+			"This command will perform the following changes:\n"+
+			"  - upgrade the `_format_version` setting to `3.0`\n"+
+			"  - add the `~` prefix to all routes' paths containing a regex-pattern\n\n"+
+			"These changes may not be correct or exhaustive enough,\n"+
+			"please perform manual audits of the converted file before applying\n"+
+			"the configuration in production.\n\n"+
+			"For more information about this and related changes,\n"+
+			"please visit: https://docs.konghq.com/deck/latest/3.0-upgrade", formatVersion)
```

Comment:
```
Error: cannot apply '1.1' config format version to Kong version 3.0 or above. 
Please upgrade your configuration to account for 3.0 breaking changes using the
following command:

deck convert --from kong-gateway-2.x --to kong-gateway-3.x

This command performs the following changes:
  - upgrade the `_format_version` value to `3.0`
  - add the `~` prefix to all routes' paths containing a regex-pattern

These changes may not be correct or exhaustive enough.
It is strongly recommended to perform perform a manual audit
of the updated configuration file before applying
the configuration in production. Incorrect changes will result in
unintended traffic routing by Kong Gateway.

For more information about this and related changes,
please visit: https://docs.konghq.com/deck/latest/3.0-upgrade

```

Please adjust line length. My edits are to make the message a bit condensed as well as make the warning a bit more strongly worded.

Reply:
https://github.com/Kong/deck/pull/753/commits/cfbf340273f6fd50cc39312fbc958f6903b7da96